luigi/contrib/hive.py

```diff
@@ -342,15 +347,19 @@ def run_job(self, job, tracking_url_callback=None):
                 query = query.encode('utf8')
             f.write(query)
             f.flush()
-            arglist = [load_hive_cmd(), '-f', f.name]
+            username_cmd = ""
+            if self.username:
+                username_cmd = "HADOOP_USER_NAME=" + self.username
+            arglist = [username_cmd] + [load_hive_cmd(), '-f', f.name]
```

Comment:
No, you are correct, this does not properly set the environment variable.
This has to be done in the call to popen, so this will need to be
refactored, unfortunately.

2015-11-25 5:29 GMT+01:00 Arash Rouhani <notifications@github.com>:

> In luigi/contrib/hive.py
> <https://github.com/spotify/luigi/pull/1426#discussion_r45828608>:
>
> > @@ -342,15 +347,19 @@ def run_job(self, job, tracking_url_callback=None):
> >                  query = query.encode('utf8')
> >              f.write(query)
> >              f.flush()
> > -            arglist = [load_hive_cmd(), '-f', f.name]
> > +            username_cmd = ""
> > +            if self.username:
> > +                username_cmd = "HADOOP_USER_NAME=" + self.username
> > +            arglist = [username_cmd] + [load_hive_cmd(), '-f', f.name]
>
> Does this really work? I always thought the environment variable setting
> in MY_VAR=my_val echo hi was done by the shell.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/luigi/pull/1426/files#r45828608>.
>